From 78d4b578a13db7512541ba1c104222dcf700a6ad Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Wed, 3 Sep 2025 10:40:31 +0200
Subject: [PATCH 886/954] refactor(st-crypto): add a crypto common header
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Some constants can be used by multiple drivers.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 985a377a9ca4ac95b71577d0d7f79af0ee46ea3f refactor(st-crypto): add a crypto common header]
Change-Id: Ie9e19ca9f8ab822c464d65490f8e885433e95f25
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/486685
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/st/crypto/stm32_crypto_common.h | 26 +++++++++++++++++++++++++
 drivers/st/crypto/stm32_saes.c          |  9 +--------
 2 files changed, 27 insertions(+), 8 deletions(-)
 create mode 100644 drivers/st/crypto/stm32_crypto_common.h

diff --git a/drivers/st/crypto/stm32_crypto_common.h b/drivers/st/crypto/stm32_crypto_common.h
new file mode 100644
index 0000000000..3b3498da69
--- /dev/null
+++ b/drivers/st/crypto/stm32_crypto_common.h
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2025, STMicroelectronics - All Rights Reserved
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#ifndef STM32_CRYPTO_COMMON_H
+#define STM32_CRYPTO_COMMON_H
+
+#include <lib/utils_def.h>
+
+/*
+ * Crypto algorithm common macro used in stm32_saes and stm32_cryp driver
+ */
+#define UINT8_BIT			8U
+#define AES_BLOCK_SIZE_BIT		128U
+#define AES_BLOCK_SIZE			(AES_BLOCK_SIZE_BIT / UINT8_BIT)
+#define AES_BLOCK_NB_U32		(AES_BLOCK_SIZE / sizeof(uint32_t))
+#define DES_BLOCK_SIZE_BIT		U(64)
+#define DES_BLOCK_SIZE			(DES_BLOCK_SIZE_BIT / UINT8_BIT)
+#define DES_BLOCK_NB_U32		(DES_BLOCK_SIZE / sizeof(uint32_t))
+#define AES_KEYSIZE_128			U(16)
+#define AES_KEYSIZE_192			U(24)
+#define AES_KEYSIZE_256			U(32)
+#define AES_IVSIZE			U(16)
+
+#endif /* STM32_CRYPTO_COMMON_H */
\ No newline at end of file
diff --git a/drivers/st/crypto/stm32_saes.c b/drivers/st/crypto/stm32_saes.c
index 147c9a1b4d..3eb02d84b2 100644
--- a/drivers/st/crypto/stm32_saes.c
+++ b/drivers/st/crypto/stm32_saes.c
@@ -23,14 +23,7 @@
 
 #include <platform_def.h>
 
-#define UINT8_BIT			8U
-#define AES_BLOCK_SIZE_BIT		128U
-#define AES_BLOCK_SIZE			(AES_BLOCK_SIZE_BIT / UINT8_BIT)
-
-#define AES_KEYSIZE_128			16U
-#define AES_KEYSIZE_192			24U
-#define AES_KEYSIZE_256			32U
-#define AES_IVSIZE			16U
+#include "stm32_crypto_common.h"
 
 #define SAES_TIMEOUT_US			100000U
 #define TIMEOUT_US_1MS			1000U
-- 
2.53.0

